R/LORDstar.R

```diff
@@ -87,29 +90,51 @@
 #'
 #'
 #' @examples
-#' pval = c(2.90e-08, 0.06743, 3.51e-04, 0.00174, 0.04723,
-#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-06,
-#'         0.69274, 0.30443, 0.00136, 0.82342, 0.54757)
-#'
-#' LORDstar(pval, version='async', decision.times=seq_len(15)) # Synchronous
+#' sample.df <- data.frame(
+#' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' decision.times = seq_len(15) + 1)
+#'
+#' LORDstar(sample.df, version='async')
 #' 
-#' LORDstar(pval, version='async', decision.times=seq_len(15)+1) # Asynchronous
+#' sample.df2 <- data.frame(
+#' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' lags = rep(1,15))
 #' 
-#'
-#' LORDstar(pval, version='dep', lags=rep(0,15)) # Synchronous
+#' LORDstar(sample.df2, version='dep')
 #' 
-#' LORDstar(pval, version='dep', lags=rep(1,15)) # Locally dependent
-#'
-#'
-#' LORDstar(pval, version='batch', batch.sizes=rep(1,15)) # Synchronous
+#' sample.df3 <- data.frame(
+#' #' id = c('A15432', 'B90969', 'C18705', 'B49731', 'E99902',
+#'     'C38292', 'A30619', 'D46627', 'E29198', 'A41418',
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757))
 #' 
-#' LORDstar(pval, version='batch', batch.sizes=c(4,6,5)) # Batched
+#' LORDstar(sample.df3, version='batch', batch.sizes = c(4,6,5))
 #'
 #' @export
 
-LORDstar <- function(pval, alpha=0.05, version, gammai, w0,
-                decision.times, lags, batch.sizes) {
+LORDstar <- function(d, alpha=0.05, version, gammai, w0, batch.sizes) {
 
+    if(is.data.frame(d)){
+        checkSTARdf(d, version)
+        pval <- d$pval
+    } else if(is.vector(d)){
+        pval <- d
```

Comment:
Add the following check: if version = 'async' or 'dep', then d should be a dataframe with a column of decision.times or lags respectively

Reply:
Done.